File: 12-03-18/python/02-sumits-happiness.py
for _ in range( int(input()) ):
    n = int(input())
    ar = [int(x) for x in input().split()]
    
    ans = 0
    psum = 0
    nsum = 0
    cnt = 0

    for i in range(0,n):
        if(ar[i]<=0):
            nsum+=ar[i]
    
    ar.sort()
    
    for i in range(n-1,0,-1):
        if(ar[i]>=0):
            psum+=ar[i]
            cnt+=1
        else:
            break
    
    ans = (psum*cnt)+nsum
    
    for i in range(n-1-cnt,0,-1):
        psum += ar[i]
        nsum -= ar[i]
        cnt += 1
        temp = (psum*cnt)+nsum
        ans = max(ans,temp)
        
    print(ans)
        

File: 12-03-18/c++/02-sumits-happiness.cpp
#include<bits/stdc++.h>

using namespace std;

typedef long long ll;


int main()
{
    int t;
    cin>>t;
    while(t--)
    {
        ll n;
        cin>>n;
        ll ar[n];
        ll nsum=0;
        for(int i=0;i<n;i++)
        {
            cin>>ar[i];
            if(ar[i]<=0)
                nsum+=ar[i];
        }
        
        sort(ar,ar+n);
        ll psum=0,ans,cnt=0;
        
        for(int i=n-1;i>=0;i--)
        {
            if(ar[i]>=0)
            {
                psum+=ar[i];
                cnt++;
            }
            else
            {
                break;
            }
        }
        ans=(psum*cnt)+nsum;
        
        for(int i=n-1-cnt;i>=0;i--)
        {
            psum+=ar[i];
            nsum-=ar[i];
            cnt++;
            ll tmp=(psum*cnt)+nsum;
            ans=max(ans,tmp);
        }
        
        cout<<ans<<endl;
        
    }
    
}  
